### DISCUSSION TITLE
How to view the 2fa secret key (not the otp)

### DISCUSSION THREAD

**Discussion by eepyfemboi** [2fast-team/2fast#86] [Q&A]
I need to view the 2fa secret keys stored in the data file in cleartext. Is there any way to do this in the app, and if not, what method could I use to decrypt the data file?

**Comment by jp-weber** [comment]
Currently this is only possible in debug mode for security reasons. You also need to convert the byte array back to a string in the source code and output or display the list to make it readable.

The crucial point is deserialization:
https://github.com/2fast-team/2fast/blob/master/Project2FA.Shared/Services/DataService.cs#L398
